### PR TITLE
[Patterns] Query resolver for Get Patterns

### LINKED ISSUE #4881: [API] GraphQL: Create Pattern List Subscription
#### Current Behavior
No subscription is available for a list of patterns.

#### Desired Behavior
Create a new subscription based on the design outlined in [Meshery API Endpoints](https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399) sheet.

#### Implementation
New subscription name: `subscribePatternsList`

#### Acceptance Tests
1. The [GraphQL API section](https://docs.meshery.io/reference/graphql-apis) of Meshery Docs should reflect this new subscription.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by bariqhibat**
**Description**

This PR fixes #4881

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: internal/graphql/resolver/pattern.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+	if err != nil {
+		r.Log.Error(err)
+		return nil, err
+	}
+
+	mc := handlers.NewContentModifier(tokenString, provider, prefObj, user.UserID)
+	err = mc.AddMetadataForPatterns(&resp)
+	if err != nil {
+		fmt.Println("Could not add metadata about pattern's current support ", err.Error())
```
Comment (leecalcote) [inline internal/graphql/resolver/pattern.go:28]: This error and the error on line 21... do these need to be MeshKit-style errors?

File: internal/graphql/schema/schema.graphql [hunk 2]
```diff
@@ -241,6 +241,35 @@ type NameSpace {
...
+	user_id: String!
+	location: PatternLocation!
+	pattern_file: String!
+	canSupport: Boolean!
+	errmsg: String
+	created_at: String
+	updated_at: String
+}
+
+type PatternLocation {
```
Comment (leecalcote) [inline internal/graphql/schema/schema.graphql:266]: Is "org" captured in here somewhere already?

**Comment by leecalcote** [comment]
Does this include an update to the [GraphQL API section](https://docs.meshery.io/reference/graphql-apis) of Meshery Docs?

**Comment by leecalcote** [comment]
Is the subscription included in this PR (is it implicit to fetch)?

**Comment by bariqhibat** [comment]
> Does this include an update to the [GraphQL API section](https://docs.meshery.io/reference/graphql-apis) of Meshery Docs?

We'll need to update the docs once this is merged

> Is the subscription included in this PR (is it implicit to fetch)?

No, subscription resolver hasn't been introduced. My plan will be to introduce it after this PR is merged since it's like an advanced version of the usual fetch

File: internal/graphql/resolver/pattern.go (hunk 1 above)
Comment (bariqhibat) [reply internal/graphql/resolver/pattern.go:28]: Yeah.. but currently we used this error in the REST counterpart

File: internal/graphql/schema/schema.graphql (hunk 2 above)
Comment (bariqhibat) [reply internal/graphql/schema/schema.graphql:266]: I don't think that we have `org` field even in the REST API response

File: internal/graphql/resolver/pattern.go (hunk 1 above)
Comment (bariqhibat) [reply internal/graphql/resolver/pattern.go]: Fixed this

File: internal/graphql/schema/schema.graphql (hunk 2 above)
Comment (leecalcote) [reply internal/graphql/schema/schema.graphql:266]: Ok, it must be captured under "host", I'm guessing.

**Comment by leecalcote** [comment]
Thank you, @bariqhibat and a quick update to the API endpoints sheet, too.

**Review by leecalcote — APPROVED** [review]

**Pull request merged**